### PR TITLE
Update to Debian 9

### PR DESCRIPTION
**Pull request by rafalbiegacz**

### CODE REVIEW

File: 1/1.12/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,44 @@
-FROM gcr.io/google-appengine/debian8
+FROM gcr.io/google-appengine/debian9
 
 ENV NGINX_VERSION 1.12.2*
 
-RUN apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-keys 573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62 \
-	&& echo "deb http://nginx.org/packages/debian/ jessie nginx" >> /etc/apt/sources.list \
+# grab gosu for easy step-down from root
+ENV GOSU_VERSION 1.10
```
Comment (jprzychodzen) [inline 1/1.12/Dockerfile]: Do we need/use gosu in this repo?

**Review by jprzychodzen — CHANGES_REQUESTED** [review]

File: 1/1.12/Dockerfile (hunk 1 above)
Comment (rafalbiegacz) [reply 1/1.12/Dockerfile]: No, we don't. I will remove it in the next version.

File: versions.yaml [hunk 2]
```diff
@@ -46,6 +50,7 @@ versions:
       - '1.12'
     from: *from
     packages:
+      gosu: *gosu
```
Comment (jprzychodzen) [inline versions.yaml]: If we do not need GOSU, please remove it from version.yaml

**Review by jprzychodzen — CHANGES_REQUESTED** [review]

File: templates/Dockerfile.template [hunk 3]
```diff
@@ -5,23 +5,44 @@ FROM {{ .From }}
...
+                apt-get install -y --no-install-recommends \
+                        gnupg \
+                        dirmngr \
+                ; \
+                apt-get install -y wget; \
+                rm -rf /var/lib/apt/lists/*; \
+        fi
+
+RUN set -x \
+{{- if eq .Dir "1/debian9/1.13" }}
```
Comment (jprzychodzen) [inline templates/Dockerfile.template:20]: Please follow up with changing this if condition to custom key.

**Review by jprzychodzen — APPROVED** [review]

**Pull request merged**